src/Extensions/ExtHostClient.rei

```diff
@@ -1,8 +1,112 @@
+module Core = Oni_Core;
 module Protocol = ExtHostProtocol;
 module Workspace = Protocol.Workspace;
-module Core = Oni_Core;
 
-type t = ExtHostTransport.t;
+type t;
+
+// SCM
+
+module SCM: {
+  // MODEL
+
+  [@deriving show]
+  type command = {
+    id: string,
+    title: string,
+    arguments: list(Core.Json.t),
+  };
```

Comment:
I'm not really sure where this belongs. In vscode it's in a module called `modes`, along with a lot of other "dumb" data structures, but that name tells me pretty much nothing of its purpose or scope.